source/support/rotate-secrets/index.html.md.erb

```diff
@@ -0,0 +1,180 @@
+weight: 35
+last_reviewed_on: 2021-01-19
+review_in: 6 months
+---
+
+# Rotate a secret
+
+If a secret expires or is compromised, you will need to rotate it.
+This is a currently a manual process.
+
+This content explains:
+
+- how secrets are used
+- what secrets we have
+- how to rotate secrets if needed
+
+If a secret is compromised, this is a security incident. Follow [the
+incident management process][] as well as rotating the secret.
+
+[the incident management process]: /incident
+
+## Prerequisites
+
+You must [install the gds-cli tool][] to manage secrets.
+
+[gds-cli]: https://github.com/alphagov/gds-cli#installation
+
+## Concourse secrets
+
+All of our secrets are used by our Concourse pipelines, and are stored
+as Concourse secrets.
+
+Some secrets are used by the pipeline to perform its task. For example,
+[the daily statistics task][] needs an OAuth token.
+
+Some secrets are used by an app and are set by the app's deployment task. For example, the
+[GOV.UK Notify API key][].
+
+Secrets are referenced in the Concourse configuration using `((secret
+name))`. For example with the daily statistics task, the
+`report-bearer-token` is a secret made available to the task through
+the `BEARER_TOKEN` environment variable:
+
+```yaml
+- task: trigger-bigquery-export
+  file: git-main/concourse/tasks/trigger-bigquery-export.yml
+  params:
+    CDN_DOMAIN: account.publishing.service.gov.uk
+    BEARER_TOKEN: ((report-bearer-token))
+```
+
+To list all secrets for the GOV.UK Concourse team, run the following in the gds-cli:
+
+```
+$ gds cd secrets ls cd-govuk-tools
+```
+
+[the daily statistics task]: https://github.com/alphagov/govuk-account-manager-prototype/blob/7e69b8cb8daec1e6e23d32374df3f8a1fa45b477/concourse/pipeline.yml#L249-L271
+[GOV.UK Notify API key]: https://github.com/alphagov/govuk-account-manager-prototype/blob/5443585c3d2dd5d399efc18f126f0bdf831036b5/concourse/tasks/deploy-to-govuk-paas.yml#L59
+
+### Rotating a secret
+
+1. Generate a new secret value, which varies per secret.
+
+2. Set the new value using the gds-cli:
+
+    ```
+    $ gds cd secrets add cd-govuk-tools <pipeline-name>/<secret-name> "<value>"
+    ```
+
+3. If this is a secret used by an app, deploy the app.
+
+## How to rotate the secrets we have
+
+In general you rotate secrets by generating a new value and
+setting the value with the gds-cli. You must use a different method for the following secrets.
+
+### Secrets used by multiple pipelines
+
+#### `grafana-api-key`
+
+Generate a new API key in [Grafana][], and deactivate the old API key.
+
+[Grafana]: https://grafana-paas.cloudapps.digital/org/apikeys
+
+#### `paas-username` and `paas-password`
+
+Change the password by [triggering a password reset][], which gets sent to the govuk-accounts-developers Google Group.
+
+[triggering a password reset]: https://admin.london.cloud.service.gov.uk/password/request-reset
+
+#### `sentry-dsn`
+
+Generate a new DSN from the Sentry project settings page.
```

Comment:
Where is the Sentry project settings page?

Reply:
Added some steps